### PR TITLE
Handle case where conflict manifests as write failure on slave

### PR DESCRIPTION
**Pull request by tylerkaraszewski**
@quinthar @cead22 @coleaeason 

This change updates a couple things.

First, we want to handle the case where we see unique constraints violations on slaves during replication. This was caused when two transactions ran in parallel on master, and both generated the same primary key. Because they were parallel, the `INSERT` queries wouldn't fail, only the final commit on the second transaction.

So, on worker threads, this wasn't a problem, because worker threads won't ever schedule a commit for replication until it's successfully committed, so the second transaction would be backed out before ever being sent to slaves.

But the sync thread doe things differently, to support quorum requests - it sends a `BEGIN_TRANSACTION` message with a query to run to slaves. This has already succeeded on master, but will fail on commit as a conflict. But master can't commit it until slaves approve it, so it sends them the transaction. When they ran the query, it failed. They'd already received and committed the first transaction (the one that this transaction conflicted with on master), so they don't see a "conflict", they see a `unique constraint violation`.

This case never happened before, so the handling for it was to crash. Now, instead of handling it, we send a `DENY_TRANSACTION` message. This is a new case, we only ever had `APPROVE_TRANSACTION` before. If master hears back from peers with `DENY_TRANSACTION`, it rolls the transaction back. This is what it would have done anyway in the case of a conflict, but this saves even having to attempt the commit when we know it will fail.

A second, smaller, part of this change is to move setting the `_commitConsistency` before we issue a `BEGIN_TRANSACTION`. This allows us to mark the transaction as ASYNC, which allows slaves to omit sending a response approving or denying a transaction back to master, which will ignore it anyway.

Finally, the test infrastructure has been updated to use a command that mimics the one that caused the unique constraints violation in the first place.

### CODE REVIEW

File: sqlitecluster/SQLiteNode.cpp [hunk 1]
```diff
@@ -686,9 +686,9 @@ bool SQLiteNode::update() {
                         numFullResponded++;
                         numFullApproved += SIEquals(response, "approve");
                         if (!SIEquals(response, "approve")) {
-                            SWARN("Peer '" << peer->name << "' declined transaction.");
+                            SWARN("Peer '" << peer->name << "' denied transaction.");
```
Comment (quinthar) [inline sqlitecluster/SQLiteNode.cpp]: Even a single denial should trigger a rollback, right?

Comment (tylerkaraszewski) [inline sqlitecluster/SQLiteNode.cpp]: Yeah, that's reasonable. If it fails on any slave we're not going to be able to safely get it onto the whole cluster.

**Comment by tylerkaraszewski** [comment]
>  This seems to assume that it's always safe to try to commit a transaction slaves have denied, on the assumption that the master would also detect a conflict when it tries to commit.

We don't do anything new specifically in response to a `deny`, because the course of action is the same as a failure to `approve`, but we do need to know that a slave has responded one way or another. There was already code in place to handle "everyone responded but not enough of them approved the transaction", but it was impossible to actually reach, because we'd never send a response that wasn't `approve`.

I'll change it to roll back on *any* denial, and that will probably make this case clearer.

**Comment by tylerkaraszewski** [comment]
@quinthar updated.

**Pull request merged**